pkg/converter/policy_converter_test.go

```diff
@@ -292,4 +299,85 @@ var _ = Describe("PolicyConverter", func() {
 			Expect(key).To(Equal(policyName))
 		})
 	})
+
+	Context("NetworkPolicy with Egress rule", func() {
+		port80 := intstr.FromInt(80)
+		np := extensions.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testPolicy",
+				Namespace: "default",
+			},
+			Spec: extensions.NetworkPolicySpec{
+				PodSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"label":  "value",
+						"label2": "value2",
+					},
+				},
+				Egress: []extensions.NetworkPolicyEgressRule{
+					{
+						Ports: []extensions.NetworkPolicyPort{
+							{Port: &port80},
+						},
+						To: []extensions.NetworkPolicyPeer{
+							{
+								PodSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"k":  "v",
+										"k2": "v2",
+									},
+								},
+							},
+						},
+					},
+				},
+				PolicyTypes: []extensions.PolicyType{extensions.PolicyTypeEgress},
+			},
+		}
+
+		// Parse the policy.
+		pol, err := npConverter.Convert(&np)
+		It("should not generate a conversion error", func() {
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		// Assert policy name.
+		It("should return calico policy with expected name", func() {
+			Expect(pol.(api.Policy).Metadata.Name).To(Equal("knp.default.default.testPolicy"))
+		})
+
+		// Assert policy order.
+		It("should return calico policy with correct order", func() {
+			Expect(int(*pol.(api.Policy).Spec.Order)).To(Equal(1000))
+		})
+
+		// Check the selector is correct, and that the matches are sorted.
+		It("should return calico policy with correct selector", func() {
+			Expect(pol.(api.Policy).Spec.Selector).To(Equal(
+				"calico/k8s_ns == 'default' && label == 'value' && label2 == 'value2'"))
+		})
+
+		protoTCP := numorstring.ProtocolFromString("tcp")
+		It("should return calico policy with correct egress rules", func() {
+			Expect(pol.(api.Policy).Spec.EgressRules).To(ConsistOf(api.Rule{
+				Action:      "allow",
+				Protocol:    &protoTCP, // Defaulted to TCP.
+				Destination: api.EntityRule{Selector: "calico/k8s_ns == 'default' && k == 'v' && k2 == 'v2'",
```

Comment:
@caseydavenport - can you confirm this conversion looks correct to you? Its a little different from the corresponding ingress rule (Ports is swapped to the Src).

Reply:
Ports should always be in the destination.

Selectors should be 
- In the "source" when doing ingress rules.
- In the "destination" when doing egress rules.